### PR TITLE
Fix position:sticky which causes wrong document positioning

### PR DESCRIPTION
**Pull request by NICEXAI**
position:sticky causes the page to fail to be positioned by the '#' symbol, so the problem is solved by adding a normal element with zero height.

### CODE REVIEW

File: docs/reference.html [hunk 1]
```diff
@@ -924,8 +924,8 @@ <h3 id='map-method'>Methods</h3>
 
 </section><section class='collapsable'>
 
-<h4 id='map-methods-for-layers-and-controls'>Methods for Layers and Controls</h4>
-
+<h4>Methods for Layers and Controls</h4>
+<div class="ghost-pointer" id='map-methods-for-layers-and-controls'></div>
```
Comment (Malvoz) [inline docs/reference.html:928]: There are some implications with "ghost pointers":

- I think search engines use the names of elements with an ID in their search results:

  docs

  Will this change affect the search results in search engines?

- If an empty `<div>` is focused nothing will be announced to screen reader users. And yes, blind users are also developers. The creator of the [Audiom](https://audiom.xrnavigation.io/) <q>cross-sensory map component</q> which is based on Leaflet was created by a blind user.

Comment (NICEXAI) [reply docs/reference.html:928]: Thank you very much for your reply.

Question 1: The principle of Soso engine is to index the keywords of the page and then search, I think this change will not affect the normal search.
Question 2: I think you are right, maybe we can add the same content in the div and hide it, what do you think?

**Comment by NICEXAI** [comment]
In any case, I think this issue of mislocation is necessary to fix, and it has caused me some inconvenience in my use of the documentation.

File: docs/reference.html (hunk 1 above)
Comment (Malvoz) [reply docs/reference.html:928]: Actually, the docs may have accessibility problems already, I haven't looked into it much yet. But we shouldn't introduce new problems.

> maybe we can add the same content in the div and hide it

Is it necessary to duplicate the content? Shouldn't CSS changes be sufficient to fix the issue? Perhaps modify the margins, scroll-margin, scroll-padding or other related properties?

Comment (NICEXAI) [reply docs/reference.html:928]: I've checked the information and haven't found a better way to solve this problem yet, unless we drop position:sticky.

Comment (NICEXAI) [reply docs/reference.html:928]: Positioning the document accurately is probably more important to me than position:sticky.

Comment (NICEXAI) [reply docs/reference.html:928]: Is there a better way to solve this problem from your side?

Comment (Malvoz) [reply docs/reference.html:928]: So there seem to be 2 problems with the sticky headings:
- Clicking a link in the toc does not always scroll to the heading / top of that section
- Even when the heading is scrolled to, there's a too large gap due to:
  https://github.com/Leaflet/Leaflet/blob/48a38c797b19cefb28002f7d09199d947260106a/docs/docs/css/main.css#L13

/cc @avioli who introduced sticky headings in https://github.com/Leaflet/Leaflet/pull/7703.

Comment (NICEXAI) [reply docs/reference.html:928]: Current information indicates that position:sticky will cause the page to not find the ID correctly.

Comment (NICEXAI) [reply docs/reference.html:928]: So far it seems that only the first problem is encountered on my side for the time being.

Comment (Malvoz) [reply docs/reference.html:928]: `reference.html` is generated from `npm run docs`, so if I understand it correctly this file shouldn't be modified directly in most cases. But someone else can confirm that.

Comment (NICEXAI) [reply docs/reference.html:928]: Yes, I think you are right, I'll think about it and see if there is a better way to solve this problem I'm having.

**Comment by avioli** [comment]
Actually the issue is because the sticky elements are sticking to the top and are not pushed off-screen, which is caused because of all headings being in one container.

Current:

https://user-images.githubusercontent.com/524259/152708451-782c52b8-6f9c-44d1-b566-44ac58e23116.mov

Possibly improved:

https://user-images.githubusercontent.com/524259/152708428-cb37d3c7-54c7-425c-ac19-f6cf98dafb50.mov

The DOM structure for the above cases is:

Current:

Screen Shot 2022-02-07 at 11 32 18 am

Possibly improved:

Screen Shot 2022-02-07 at 11 32 47 am

Following is a snippet I've came up with as a proof - to wrap all elements after the `toc` within a `div.content-item` and add the necessary style overrides:

(You can copy/paste it in your web-dev console, then close the console for the aside navigation to actually work)

```js
(() => {
  let collect = false;
  let collected = [];
  const groups = [];
  document.querySelectorAll('div.container > *').forEach(el => {
    if (
      collected.length &&
      el.nodeName === 'A' &&
      el.getAttribute('anchor')
    ) {
      el.id = el.getAttribute('anchor');
      groups.push(collected);
      collected = [];
    }
    if (el.nodeName === 'H2' && el.id) {
      el.id = null;
    }
    if (collect) {
      collected.push(el);
    } else {
      collect = el.id === 'toc';
    }
  });
  if (collected.length) {
    groups.push(collected);
  }
  groups.forEach(group => {
    const div = document.createElement('div');
    div.className = 'content-item';
    group[0].insertAdjacentElement('beforebegin', div);
    group.forEach(item => div.appendChild(item));
  });
  const style = document.createElement('style');
  style.textContent = `
  html {
    scroll-padding-top: 0;
  }
  .content-item {
    padding-top: 2em;
  }

  /* added */
  .api-page #toc ~ .content-item > h2,
  /* copied */
  .api-page #toc ~ h2 {
    font-weight: 700;
    font-size: 1.75em;
    color: #333;
    border-bottom: 3px solid #555;
    transition: border-color .25s ease;
    position: sticky;
    top: 0;
    z-index: 1;
    background-color: white;
    /* added */
    margin: 0;
  }

  /* added */
  .api-page #toc ~ .content-item > h2 ~ * {
    margin-top: 1.5em;
  }
  `;
  document.querySelector('head').appendChild(style);
})();
```

As you may have seen in the snippet - the `H2` ids are removed and "put" to the anchors before them, but I'm unsure of the implication for accessibility. I also have a feeling these anchors are generated by JavaScript, but I might be wrong.

Unfortunately the Map Methods and Map Misc are still not positioned correctly.

File: docs/reference.html (hunk 1 above)
Comment (avioli) [reply docs/reference.html:928]: I added a comment to the main thread of the PR

**Comment by NICEXAI** [comment]
> Actually the issue is because the sticky elements are sticking to the top and are not pushed off-screen, which is caused because of all headings being in one container.
> 
> Current:
> 
>  Screen.Recording.2022-02-07.at.11.28.34.am.mov 
> Possibly improved:
> 
>  Screen.Recording.2022-02-07.at.11.27.38.am.mov 
> The DOM structure for the above cases is:
> 
> Current:
> 
> Screen Shot 2022-02-07 at 11 32 18 am
> 
> Possibly improved:
> 
> Screen Shot 2022-02-07 at 11 32 47 am
> 
> Following is a snippet I've came up with as a proof - to wrap all elements after the `toc` within a `div.content-item` and add the necessary style overrides:
> 
> (You can copy/paste it in your web-dev console, then close the console for the aside navigation to actually work)
> 
> ```js
> (() => {
>   let collect = false;
>   let collected = [];
>   const groups = [];
>   document.querySelectorAll('div.container > *').forEach(el => {
>     if (
>       collected.length &&
>       el.nodeName === 'A' &&
>       el.getAttribute('anchor')
>     ) {
>       el.id = el.getAttribute('anchor');
>       groups.push(collected);
>       collected = [];
>     }
>     if (el.nodeName === 'H2' && el.id) {
>       el.id = null;
>     }
>     if (collect) {
>       collected.push(el);
>     } else {
>       collect = el.id === 'toc';
>     }
>   });
>   if (collected.length) {
>     groups.push(collected);
>   }
>   groups.forEach(group => {
>     const div = document.createElement('div');
>     div.className = 'content-item';
>     group[0].insertAdjacentElement('beforebegin', div);
>     group.forEach(item => div.appendChild(item));
>   });
>   const style = document.createElement('style');
>   style.textContent = `
>   html {
>     scroll-padding-top: 0;
>   }
>   .content-item {
>     padding-top: 2em;
>   }
> 
>   /* added */
>   .api-page #toc ~ .content-item > h2,
>   /* copied */
>   .api-page #toc ~ h2 {
>     font-weight: 700;
>     font-size: 1.75em;
>     color: #333;
>     border-bottom: 3px solid #555;
>     transition: border-color .25s ease;
>     position: sticky;
>     top: 0;
>     z-index: 1;
>     background-color: white;
>     /* added */
>     margin: 0;
>   }
> 
>   /* added */
>   .api-page #toc ~ .content-item > h2 ~ * {
>     margin-top: 1.5em;
>   }
>   `;
>   document.querySelector('head').appendChild(style);
> })();
> ```
> 
> As you may have seen in the snippet - the `H2` ids are removed and "put" to the anchors before them, but I'm unsure of the implication for accessibility. I also have a feeling these anchors are generated by JavaScript, but I might be wrong.
> 
> Unfortunately the Map Methods and Map Misc are still not positioned correctly.

Nice, but he only seems to solve the problem of title overlays, not positioning errors.

**Pull request closed without merging**